src/Nethermind/Nethermind.TxPool/Filters/TooExpensiveTxFilter.cs

```diff
@@ -45,19 +48,43 @@ public TooExpensiveTxFilter(IChainHeadInfoProvider headInfo, IAccountStateProvid
             IReleaseSpec spec = _specProvider.GetSpec();
             Account account = _accounts.GetAccount(tx.SenderAddress!);
             UInt256 balance = account.Balance;
-            UInt256 affordableGasPrice = tx.CalculateAffordableGasPrice(spec.IsEip1559Enabled, _headInfo.CurrentBaseFee, balance);
+            UInt256 cumulativeCost = UInt256.Zero;
+            bool overflow = false;
 
-            bool overflow = spec.IsEip1559Enabled && UInt256.AddOverflow(tx.MaxPriorityFeePerGas, tx.MaxFeePerGas, out _);
+            Transaction[] transactions = _txs.GetBucketSnapshot(tx.SenderAddress);
+
+            for (int i = 0; i < transactions.Length; i++)
+            {
+                if (transactions[i].Nonce < tx.Nonce)
```

Comment:
Not sure if we want to count nonce gaps? Won't they be deprioritized by GasBottleneck?

Reply:
In full txPool gap-nonces are first to evict - we can have gap-nonces only if out txPool capacity is higher than number of 'ready' transactions (as 'ready' i mean with next-nonce-to-execute or when we already have tx with previous nonce).

I think we can consider not counting cumulativeCost at all when txPool is not full